### ISSUE TITLE
Color matching doesn't work

### ISSUE THREAD

**Issue by Billtholomew** [Billtholomew/SetPlayer#1]
Program sees what should be the same color as different colors. This may be an issue with getting the actual color itself. It may be an issue, largely, with the "line" shading. Need to double check if this is also an issue with "empty" shading.

**Comment by Billtholomew** [comment]
Partial fix is now in place.
Converting to HSV, maxing out S and V to get only H as a variable, then converting back to RGB gives us a basic "ROYGBIV" color to work with, which is more or less equivalent to what we want. By comparing which of R,G,B is maxed out (which is the result of maxing out S and V) we can isolate which "color area" we are in. This helps reduce the error caused by a 20% deviation from one end of green may yield green in one direction, but blue in the other.
Next part of this to resolve is to find how close we are in these "color areas."

**Comment by Billtholomew** [comment]
Color matching issue has been resolved.

Using the hue value from HSV, we get a value from 0-359. This is how it was kind of working before. The question became, how do we easily tell that 355 is close to 5?
The simple solution is to recall that each of these can be represented by a point on a circle. Just like the unit circle. So we change the theta to a x and y value and now it's easy to check distances!

Note that we also updated this so we don't compare attributes to each other any more. Now we try to "learn" what the different attributes (such as color) are using clustering. For color, using x and y works great because it helps with visualization and in making more distinct clusters.
